### PR TITLE
colorbalancergb: fix alignment to avoid crash when activating shadow/mid/highlight masks

### LINKED ISSUE #14752: Darktable crash on enabling color balance rgb mask visualization
### Describe the bug

so everytime when i try to visualize a mask in RGB calibration tool my darktable close without give me any message .

### Steps to reproduce

open darktable
go to dark room
select calibration color RGB
then go to mask and try to vizualise the mask applied by clicking in a circle figure in the right corner of the type of mask from shadow or highlight

### Expected behavior

it should appear where is the mask applied in my image 

### Logfile | Screenshot | Screencast

_No response_

### Commit

_No response_

### Where did you install darktable from?

distro packaging

### darktable version

4.4.0

### What OS are you using?

Linux

### What is the version of your OS?

ubuntu 22.04.2 LTS

### Describe your system?

_No response_

### Are you using OpenCL GPU in darktable?

None

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

_No response_

### PR DESCRIPTION
**Pull request by ralfbrown**
Fixes #14752.

### CODE REVIEW

File: src/iop/colorbalancergb.c [hunk 1]
```diff
@@ -1328,7 +1328,7 @@ void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *p1, dt_dev_pix
 
 void init_pipe(dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe, dt_dev_pixelpipe_iop_t *piece)
 {
-  piece->data = calloc(1, sizeof(dt_iop_colorbalancergb_data_t));
+  piece->data = dt_calloc_align(64, sizeof(dt_iop_colorbalancergb_data_t));
```
Comment (TurboGit) [inline src/iop/colorbalancergb.c:1331]: Maybe a corresponding `dt_free_align()` in `cleanup_pipe()`?

Comment (ralfbrown) [reply src/iop/colorbalancergb.c:1331]: Yes, you're right.

File: src/iop/colorbalancergb.c [hunk 2]
```diff
@@ -1064,7 +1064,7 @@ void cleanup_global(dt_iop_module_so_t *module)
 {
   dt_iop_colorbalancergb_global_data_t *gd = (dt_iop_colorbalancergb_global_data_t *)module->data;
   dt_opencl_free_kernel(gd->kernel_colorbalance_rgb);
-  free(module->data);
+  dt_free_align(module->data);
```
Comment (TurboGit) [inline src/iop/colorbalancergb.c:1067]: Not this one as allocated with `malloc()` in `init_global()`.

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**